spec/async/redis/context/pipeline_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'async/redis/client'
+require 'async/redis/context/pipeline'
+
+RSpec.describe Async::Redis::Context::Pipeline, timeout: 5 do
+	include_context Async::RSpec::Reactor
+
+	let(:endpoint) {Async::Redis.local_endpoint}
+	let(:client) {Async::Redis::Client.new(endpoint)}
+	let(:pool) {client.instance_variable_get(:@pool)}
+	let(:pipeline) {Async::Redis::Context::Pipeline.new(pool)}
+
+	let(:example_key_vals) do
+	  { pipeline_key_1: '123', pipeline_key_2: '456' }
+	end
+
+	describe '.call' do
+	  it 'accumulates commands without running them' do
```

Comment:
Transactions and pipelines are different.

With a `MULTI/EXEC` transaction, Redis guarantees that all the commands between the `MULTI` and `EXEC` are executed atomically.

Pipelines are different, Redis does not guarantee that all the commands run in a pipeline will be executed atomatically. There can be other Redis commands that will run betwene them. Pipelines are mostly used to avoid unnecessary network round-trips to Redis when there's no need to run the commands atomically.

Reply:
That's correct, however there is no reason why you couldn't pipeline a whole transaction with something like this.
```ruby
pipeline.multi do |c|
  #do stuff atomically
end

# pipeline more requests
pipeline.run
```
However you might be in trouble if the transaction blows up in the pipeline.